src/merge_maps_tool.cpp

```diff
@@ -179,6 +198,119 @@ bool MergeMapTool::MergeMapCallback(slam_toolbox::MergeMaps::Request &req,
+        auto bbox_min_corr = karto::Vector2<kt_double>(bbox_min_tf_corr.getOrigin().x(),bbox_min_tf_corr.getOrigin().y());
+        bbox.SetMinimum(bbox_min_corr);
+
+        tf::Transform bbox_max_tf;
+        bbox_max_tf.setOrigin(tf::Vector3(bbox.GetMaximum().GetX(),bbox.GetMaximum().GetY(),0.));
+        bbox_max_tf.setRotation(tf::createQuaternionFromRPY(0, 0, 0));
+        tf::Transform bbox_max_tf_corr;
+        bbox_max_tf_corr = submap_correction*bbox_max_tf;//*
+        auto bbox_max_corr = karto::Vector2<kt_double >(bbox_max_tf_corr.getOrigin().x(),bbox_max_tf_corr.getOrigin().y());
+
+        bbox.SetMaximum(bbox_max_corr);
+        pScan_copy->SetBoundingBox(bbox);
+
+         // TRANSFORM UNFILTERED POINTS USED
+        karto::PointVectorDouble UPR_vec = pScan_copy->GetPointReadings();
+        for(auto it_upr = UPR_vec.begin(); it_upr!=UPR_vec.end();it_upr++)
+        {
+
+          tf::Transform upr_tf;
+          upr_tf.setOrigin(tf::Vector3((*it_upr).GetX(),(*it_upr).GetY(),0.));
+          upr_tf.setRotation(tf::createQuaternionFromRPY(0, 0, 0));
+
+          tf::Transform upr_corr;
+          upr_corr = submap_correction*upr_tf;//*
+          (*it_upr).SetX(upr_corr.getOrigin().getX());
+          (*it_upr).SetY(upr_corr.getOrigin().getY());
+        }
+        pScan_copy->SetPointReadings(UPR_vec);
+
+
+        //TRANSFORM CORRECTED POSE
+        corrected_pose = pScan_copy->GetCorrectedPose();
+        tf::Transform corr_pose_tf;
+        corr_pose_tf.setOrigin(tf::Vector3(corrected_pose.GetX(), \
+                                    corrected_pose.GetY(), 0.));
+        corr_pose_tf.setRotation(tf::createQuaternionFromRPY(0, 0, corrected_pose.GetHeading()));
+
+        tf::Transform robotPose_corr;
+        robotPose_corr = submap_correction*corr_pose_tf;//*
+
+        karto::Pose2 karto_robotPose_corr = \
+            karto::Pose2(robotPose_corr.getOrigin().x(), \
+                           robotPose_corr.getOrigin().y(), tf::getYaw(robotPose_corr.getRotation()));
+        pScan_copy->SetCorrectedPose(karto_robotPose_corr);
+        kt_bool rIsDirty = false;
+        pScan_copy->SetIsDirty(rIsDirty);
+
+        //TRASNFORM ODOM POSE
+        auto odom_pose = pScan_copy->GetOdometricPose();
+        tf::Transform odom_pose_tf;
+        odom_pose_tf.setOrigin(tf::Vector3(odom_pose.GetX(), \
+                                    odom_pose.GetY(), 0.));
+        odom_pose_tf.setRotation(tf::createQuaternionFromRPY(0, 0, odom_pose.GetHeading()));
+
+        tf::Transform robotPose_odom;
+        robotPose_odom = submap_correction*odom_pose_tf;//*
+
+        karto::Pose2 karto_robotPose_odom = \
+              karto::Pose2(robotPose_odom.getOrigin().x(), \
+                           robotPose_odom.getOrigin().y(), tf::getYaw(robotPose_odom.getRotation()));
+
+        pScan_copy->SetOdometricPose(karto_robotPose_odom);
+
+        transformed_scans.push_back(pScan_copy);
+
```

Comment:
this entire chunk is way way too long. You need to break this out into multiple functions, its alot of the same boiler plate over and over again